administrator/components/com_associations/models/forms/filter_associations.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<form>
+	<fields name="filter">
+		<field
+			name="search"
+			type="text"
+			label="COM_ASSOCIATIONS_FILTER_SEARCH_LABEL"
+			description="COM_CONTENT_FILTER_SEARCH_DESC"
+			hint="JSEARCH_FILTER"
+		/>
+		<field
+			name="component"
+			type="associatedcomponent"
+			label="COM_ASSOCIATIONS_FILTER_SEARCH_LABEL"
+			description="COM_CONTENT_FILTER_SEARCH_DESC"
+			>
+			<option value="">COM_ASSOCIATIONS_FILTER_SELECT_ITEM_TYPE</option>
+		</field>
+		<field
+			name="language"
+			type="contentlanguage"
+			label="JOPTION_FILTER_LANGUAGE"
+			description="JOPTION_FILTER_LANGUAGE_DESC"
+			>
+			<option value="">JOPTION_SELECT_LANGUAGE</option>
+		</field>
+	</fields>
```

Comment:
you need to add the ordering here too

See other filter_xxx.xml files.
Example: https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_contact/models/forms/filter_contacts.xml#L83-L135